lib/temporal.rb

```diff
@@ -81,6 +83,40 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    def await_workflow_result(workflow:, workflow_id:, run_id:, **args)
```

Comment:
some of the other "follow up" methods take a `namespace` instead of a workflow, but I found it hard to use.  I wanted it to match the same logic as `start_workflow`.  It's actually harder for developers to get the namespace and replicate the logic themselves.  Would love to see if we can go back and fix the others.
This is also partially inconsistent in that it takes kwargs.  I noted that some of other methods take the same parameters in different orders and with different choices on keyword vs positional args.  just going full-on kwargs seems to be the best policy, and easier to read the callsites.